services/pull/patch.go

```diff
@@ -309,6 +309,69 @@ func CheckFileProtection(oldCommitID, newCommitID string, patterns []glob.Glob,
 	return changedProtectedFiles, err
 }
 
+// CheckUnprotectedFiles check if the commit only touches unprotected files
+func CheckUnprotectedFiles(oldCommitID, newCommitID string, patterns []glob.Glob, env []string, repo *git.Repository) (bool, error) {
```

Comment:
CheckFileProtection use similar mechanism  ...

best would be if modules/git would provide souch mechanism (to get affected files)
but in any case we could at least do some code dedub :)

Reply:
Extracted GetAffectedFiles function to modules/git.